packages/server/controllers/auth/domainBlacklist.js

```diff
@@ -1,19 +1,42 @@
-// Load blacklisted domains once
-const blacklistedDomains = process.env.LOGCHIMP_BLACKLISTED_DOMAINS?
-     new Set(process.env.LOGCHIMP_BLACKLISTED_DOMAINS.split(',').map(domain => domain.trim().toLowerCase())
+// Valid domain regex (e.g., example.com, sub.domain.org)
+const domainRegex = /^(?!-)([A-Za-z0-9-]+\.)+[A-Za-z]{2,}$/;
+
+// Load and validate blacklisted domains
+const blacklistedDomains = process.env.LOGCHIMP_BLACKLISTED_DOMAINS
+  ? new Set(
+      process.env.LOGCHIMP_BLACKLISTED_DOMAINS
+        .split(',')
+        .map(domain => domain.trim().toLowerCase())
+        .filter(domain => {
+          if (!domain) return false; // remove empty strings
+          const isValid = domainRegex.test(domain);
+          if (!isValid) {
+            console.warn(`⚠️ Invalid domain in blacklist config: "${domain}"`);
```

Comment:
Instead of doing `console.warn`, try using the `logger` util function in `packages/server/utils/logger.js` file.